### DISCUSSION TITLE
how can i use AutoLoad variables in  Dialogic？

### DISCUSSION THREAD

**Discussion by seed071608** [dialogic-godot/dialogic#1453] [Q&A] [answered]
My English is poor！！
I use Godot 3.5.1
![QQ截图20230302205301](https://user-images.githubusercontent.com/69508761/222434369-2c670692-086e-44f4-87a5-1879981e0893.png)

![QQ截图20230302205338](https://user-images.githubusercontent.com/69508761/222434400-d1ac3a5b-862a-401b-bd5b-c1d0ea0fec40.png)
The player's name is Bob, and the name is saved in autoload. How can I use the variable of autoload in the conversation

Any help is appreciated!

**Comment by zaknafean** [comment, marked as answer]
So what you need to do is use dialogic definitions to bridge between GDScript and Dialogic. This is thankfully pretty simple, but is a multistep process, so here it goes.

**Step 1** Create the definition in Dialogic. You can name this whatever you want
![image](https://user-images.githubusercontent.com/7741797/222438951-c7c81519-c936-4f6c-a0e8-cc9f85bf4a8e.png)

Feel free to give it a default value for testing
![image](https://user-images.githubusercontent.com/7741797/222439272-243c3196-0b16-432d-a1d0-d8cb048197e1.png)


**Step 2** Set the definition in GDScript. This is pretty straight forward.
```
Dialogic.set_variable('active-item-count', '5 Bananas')
```
The link has been made! Now how do you use it?

**Step 3** Use the definition in Dialogic
Very simple again just use brackets!

![image](https://user-images.githubusercontent.com/7741797/222439836-0175a547-dff9-4962-9757-ddf2d7619f32.png)

And it will look a little something like this
![image](https://user-images.githubusercontent.com/7741797/222440081-1a465b1a-e3ac-47ba-af97-01e5cba1d488.png)

You can change the value of a definition with the Set Value Event
![image](https://user-images.githubusercontent.com/7741797/222440256-b26e78d1-d2e6-4699-aa9d-4eed8a219cde.png)

And you can get the current value of a definution in GDScript with the get_variable call like so
```
Dialogic.get_variable('active-item-count')
```

WHY do we use the terms VALUE, DEFINTION, and VARIABLE to all refer to the same thing... well that's one of many reasons the Dialogic 2.x project is such a big rewrite!

**Reply by seed071608** [reply]
thank you！
